work_queue/src/bindings/python3/brett.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import work_queue as wq
+
+def wqmap(f, a):
+    q = wq.WorkQueue(9123)
+    results = []
+    for i in a:
+        p_task = wq.PythonTask(f, i)
+        q.submit(p_task)
+
+    while not q.empty():
+        t = q.wait()
+        if t:
+            x = t.output
+            if isinstance(x, wq.PythonTaskNoResult):
+                print("Task {} failed and did not generate a result.".format(t.id))
+            else:
+                results.insert(0, x)
+
+    return results
+
+def wqallpairs(f, a, b):
+    q = wq.WorkQueue(9123)
+    results = []
+    for i in a:
+        for j in b:
+            p_task = wq.PythonTask(f, i, j)
+            q.submit(p_task)
+
+    while not q.empty():
+        t = q.wait()
+        if t:
+            x = t.output
+            if isinstance(x, wq.PythonTaskNoResult):
+                print("Task {} failed and did not generate a result".format(t.id))
+            else:
+                results.insert(0, x)
+
+    return results
+
+def wqtreereduce(f, a):
+    q = wq.WorkQueue(9123)
+
+    tmp = a
+    size = len(a)
+
+    while size != 1:
+        results = []
+        for i in range(len(tmp)//2):
```

Comment:
This will loose results if `len(tmp)` is odd.

Reply:
I had a question about that - would that odd one just carry down till the end? I was not sure what to do with that

Reply:
The odd one, if any, will always be the last in `tmp`, so we can add it as the last of `results`. Eventually it will be processed in some iteration.